### PR TITLE
PD-257: Standardize git messages

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Introduce enforce standardization of commit messages to the leafy-green repo, based on conventional commits (a pre-cursor to generating an automated changelog)

🎟 _Jira ticket:_ [PD-257](https://jira.mongodb.org/browse/PD-257)

## 🛠 Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## 💬 Further comments

Brought in a package, https://www.npmjs.com/package/@commitlint/config-conventional, to act as the linter and use a husky `commit-msg` hook to trigger the linting.

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
```
Comment (hswolff) [inline commitlint.config.js]: Can we also include https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/config-lerna-scopes to lint the scopes?

Comment (brookescarlett) [reply commitlint.config.js]: Yep!

**Comment by hswolff** [comment]
We want to add notes about this in the README? To educate contributors how we require commit messages?

**Comment by brookescarlett** [comment]
@hswolff updated our documentation!

**Pull request merged**